Eastern-Research-Group/expert-query
# Feature/146 alphabetize select input lists

## Related Issues:
* [EQ-146](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-146)

## Main Changes:
* Alphabetized select input fields.
    * Database values are sorted server-side with an `ORDER BY` clause.
    * List and domain values are sorted when they are first loaded into the application's state.
* Fixed an issue where numeric fields were throwing errors when applied to `WHERE ILIKE` clauses.
* Added missing fields, corrected bad domain value aliases.
    * These domain values aren't matching the correct date types expected. I followed the batch templates provided by the EQ team as best as possible, but I'm not sure that we should rely on their domain services too much. We will need to look into the storage and time costs associated with table indices.
* added `numeric` type fields to applicable columns in the server-side column mapping. If not present, a column defaults to `varchar` or `text`. The updated raw `WHERE ILIKE` clause performs equally well for both strings and numbers, but the type properties may come in handy at a later time.

## Steps To Test:
1. Open the applications homepage: http://localhost:3000/attains/.
2. Select any profile.
3. Confirm that the input values are listed in alphabetically ascending order.

## TODO:
- [ ] This PR relies on #39
